pkg/resources/resource_source_postgres.go

```diff
@@ -95,6 +87,94 @@ func SourcePostgres() *schema.Resource {
 	}
 }
 
+func sourcePostgresRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	i := d.Id()
+
+	metaDb, region, err := utils.GetDBClientFromMeta(meta, d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	s, err := materialize.ScanSource(metaDb, utils.ExtractId(i))
+	if err == sql.ErrNoRows {
+		d.SetId("")
+		return nil
+	} else if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(utils.TransformIdWithRegion(string(region), i))
+
+	if err := d.Set("name", s.SourceName.String); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("schema_name", s.SchemaName.String); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("database_name", s.DatabaseName.String); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("size", s.Size.String); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("cluster_name", s.ClusterName.String); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("ownership_role", s.OwnerName.String); err != nil {
+		return diag.FromErr(err)
+	}
+
+	b := materialize.Source{SourceName: s.SourceName.String, SchemaName: s.SchemaName.String, DatabaseName: s.DatabaseName.String}
+	if err := d.Set("qualified_sql_name", b.QualifiedName()); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("comment", s.Comment.String); err != nil {
+		return diag.FromErr(err)
+	}
+
+	deps, err := materialize.ListPostgresSubsources(metaDb, utils.ExtractId(i), "subsource")
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// Tables
+	tMaps := []interface{}{}
+	for _, dep := range deps {
+		tMap := map[string]interface{}{}
+		tMap["name"] = dep.TableName.String
+		tMap["alias"] = dep.ObjectName.String
+		tMaps = append(tMaps, tMap)
+	}
+	if err := d.Set("table", tMaps); err != nil {
+		return diag.FromErr(err)
+	}
+
+	subs, err := materialize.ListDependencies(metaDb, utils.ExtractId(i), "source")
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// Subsources
+	sMaps := []interface{}{}
+	for _, dep := range subs {
+		sMap := map[string]interface{}{}
+		sMap["name"] = dep.ObjectName.String
+		sMap["schema_name"] = dep.SchemaName.String
+		sMap["database_name"] = dep.DatabaseName.String
+		sMaps = append(sMaps, sMap)
+	}
+	if err := d.Set("subsource", sMaps); err != nil {
+		return diag.FromErr(err)
+	}
```

Comment:
@bobbyiliev just making sure I'm understanding correctly — this logic is still to be updated, right?

Reply:
Yep this still needs to be updated to introduce the new schema_name attr for both the upstream and local tables.